### PR TITLE
Gae

### PR DESCRIPTION
**Pull request by arjun23496**
* Added provision for Generalized Advantage Estimation

### CODE REVIEW

File: pg_methods/baselines.py [hunk 1]
```diff
@@ -61,12 +62,13 @@ def __init__(self, fn_approximator, optimizer=None):
...
 
-    def update_baseline(self, trajectory, returns):
+    def update_baseline(self, trajectory, returns, tro=False):
         """
+        Function to update the baseline function approximator
 
         :param trajectory:
         :param returns:
-        :return:
+        :tro: Use Trust Region Optimization for value function update (ref: https://arxiv.org/pdf/1506.02438.pdf)
```
Comment (zafarali) [inline pg_methods/baselines.py]: :param tro:

File: pg_methods/gradients.py [hunk 2]
```diff
@@ -56,7 +83,7 @@ def calculate_policy_gradient_terms(log_probs, advantage):
     if not isinstance(advantage, Variable):
         advantage = Variable(advantage)
 
-    return -log_probs * advantage
+    return -log_probs.float() * advantage.float()
```
Comment (zafarali) [inline pg_methods/gradients.py]: Are these doubles otherwise?

Comment (arjun23496) [reply pg_methods/gradients.py]: do you mean to say that check if the input to the function is float?

Comment (zafarali) [reply pg_methods/gradients.py]: yes

**Pull request merged**